Gemfile

```diff
@@ -14,4 +14,8 @@ end
 # your gem to rubygems.org.
 
 # Load Gemfile with dependencies from manageiq
-eval_gemfile(File.expand_path("spec/manageiq/Gemfile", __dir__))
+MANAGEIQ_GEMFILE = File.expand_path("spec/manageiq/Gemfile", __dir__)
+unless File.exist?(MANAGEIQ_GEMFILE)
+  raise 'You must run `bin/setup` to install manageiq into spec/manageiq as a rails dummy app for this provider engine gem'
```

Comment:
> `bin/setup` is PORO.

Now that I look at it again, you are right, that would be without issue since you clone then `bundle` right after.  Subsequent commands following that would then have the gems install and it would work fine.

Truth be told, I think putting this either in the Gemfile or the Rakefile is fine, and your suggestion for how to do it in the Rakefile would have probably been what I would have come up with myself.  My only suggestion might be using a:

```ruby
puts "<<YOUR ERROR MESSAGE HERE>>"
exit 1
```

for the error message instead, as it will avoid extra noise from `rake`/`bundler`'s error handling mechanisms.  I assume you just want to deliver a simple message to your users that all of the dependencies haven't been installed (stacktraces can make this unclear in my opinion).

I think the only reason I shy away from putting this Gemfile is traditionally (from my point of view), I tend to see the Gemfile as a list of gems to install, and ideally nothing more.  But this is an interesting and non-standard Gem setup, and I wouldn't call myself a "foremost expert" on the subject, so take my opinion with a grain of salt.